apiserver/plane/app/views/notification/base.py

```diff
@@ -360,3 +362,81 @@ def patch(self, request):
             serializer.save()
             return Response(serializer.data, status=status.HTTP_200_OK)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+
+class WorkspaceUserNotificationPreferenceEndpoint(BaseAPIView):
+    model = UserNotificationPreference
+    serializer_class = UserNotificationPreferenceSerializer
+
+    @allow_permission(
+        allowed_roles=[ROLE.ADMIN, ROLE.MEMBER, ROLE.GUEST], level="WORKSPACE"
+    )
+    def get(self, request, slug):
+        workspace = Workspace.objects.get(slug=slug)
+        get_notification_preferences = (
```

Comment:
Line 377 to 395 can be clubbed into the following.
```
existing_transports = UserNotificationPreference.objects.filter(
    workspace=workspace, user=request.user
).values_list('transport', flat=True)
create_transports = [
    transport
    for transport, _ in NotificationTransportChoices.choices
    if transport not in existing_transports
]
```
This will be concise and improve readability of the code.